doc/proposals/ansible-operator.md

```diff
@@ -0,0 +1,78 @@
+* /operator-sdk/pkg/ansible/runner
+  * Contains the runner types and interfaces
+  * Implementation is behind an internal package (/operator-sdk/pkg/ansible/runner/internal)
+  * New Methods are exposed and are the main way a user interacts with the package
+  * Runner interface for running ansible from the operator.
+  * NewForWatchers - the method that returns a map of GVK to Runner types based on the watchers file.
+  * NewPlaybookRunner - the method that returns a new Runner for a playbook.
+  * NewRoleRunner - the method that returns a new Runner for a role.
+  * This contains the events API code and public methods. Implementation should probably be in the internal package. The events API is used for recieving events from ansible runner.
+
+* /operator-sdk/pkg/ansible/proxy
+  * This is a reverse proxy for the kubernetes API that is used for owner reference injection.
+* /operator-sdk/pkg/ansible/proxy/kubeconfig
+  * Code needed to generate the kubeconfig for the proxy.
+* /operator-sdk/pkg/ansible/events
+  * Package for event handlers from ansible runner events.
+  * Default has only the event logger.
+
+
+### Commands
+We are adding and updating existing commands to accommodate the ansible operator.  Changes to the `cmd` package as well as changes to the generator are needed.
+
+`operator-sdk new --type ansible --kind <kind> --api-version <group/version>`  This will be a new generation command under the hood. We will:
+* Create a new ansible role in the roles directory
+* Create a new watchers file. The role and GVK are defaulted based on input to the command.
+* A CRD is generated. This CRD does not have any validations defined.
+* A dockerfile is created using the watchers file and the ansible role with the base image being the ansible operator base image.
+
+The resulting structure should be
+```
+|- Dockerfile
+|
+|- roles
+\ | - <kind>
+|  \ | - generated ansible role
+|
+| - watchers.yaml
+|
+| - deploy
+\  | - <kind>-CRD.yaml
+|  | - rbac.yaml
+|  | - operator.yaml
+|  | - cr.yaml
+```
+
+`operator-sdk generate crd <api-version> <kind> ` This will be used to generate new CRDs based on ansible code. The command helps when a user wants to watch more than one CRD for their operator.
+Args:
+Required kind - the kind for the object.
+Required api-version - the <group>/<version> for the CRD.
+Flags:
+Optional: --defaults-file - A path to the defaults file to use to generate a new CRD.yaml. If this is not defined, then an empty CRD is created.
+
+`operator-sdk up local` - This should use the known structure and the ansible operator code to run the operator from this location. This will need to be changed to determine if it is an ansible operator or a golang operator. The command works by running the operator-sdk binary, which includes the ansible operator code, as the operator process. This is slightly different than the current up local command.
+
```

Comment:
> running the operator-sdk binary, which includes the ansible operator code

Sorry if I'm rehashing this again but how exactly does the operator-sdk run the user's local ansible code?
In a Golang project it would just run the main entry point from the local project ` exec.Command("go run cmd/manager/main.go")`.

And for a pure ansible project `up local` is just going to run the main function inside its own process right? i.e start a new manager, read the local watches.yaml and create a controller for each gvk.

Basically `up local` would run this internally. 
https://github.com/water-hole/ansible-operator/blob/master/cmd/ansible-operator/main.go#L29-L78

Is that right?

Reply:
In a pure ansible operator, `up local` would run the code you linked to, which would be already compiled into the `operator-sdk` binary.

In a "hybrid" operator, `up local` would just treat it like any other golang operator.

Reply:
That is correct!